### PR TITLE
Add Oracle Linux 8 based container images for PHP 7.4.

### PR DESCRIPTION
**Pull request by Djelibeybi**
Signed-off-by: Avi Miller <avi.miller@oracle.com>

### CODE REVIEW

File: OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2020 Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+FROM oraclelinux:8-slim
+
+LABEL "org.opencontainers.image.authors"="Oracle Linux Product Team <ol-ovm-info_ww@oracle.com>" \
+      "org.opencontainers.image.url"="https://github.com/oracle/docker-images" \
+      "org.opencontainers.image.source"="https://github.com/oracle/docker-images/tree/master/OracleLinuxDevelopers/oraclelinux8/php/7.4-apache" \
+      "org.opencontainers.image.vendor"="Oracle America, Inc" \
+      "org.opencontainers.image.description"="Oracle Linux 8 (slim) with PHP 7.4 and Apache 2.4.37 installed."
```
Comment (mark-au) [inline OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile:9]: Will the minor version of httpd remain constant through the module or should this be more generic 2.4.x?

File: OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                -keyout /etc/pki/tls/private/localhost.key \
+                -out /etc/pki/tls/certs/localhost.crt \
+                -days 3650 -subj '/CN=localhost' && \
+    # Redirect logging to stdout/stderr for container logging to work
+    sed -i 's/;error_log = syslog/error_log = \/dev\/stderr/' /etc/php.ini && \
+    ln -sf /dev/stdout /var/log/httpd/access_log && \
+    ln -sf /dev/stderr /var/log/httpd/error_log && \
+    # Disable userdirs and the auto-generated welcome message
+    rm -f /etc/httpd/conf.d/{userdir.conf,welcome.conf}
+
```
Comment (mark-au) [inline OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile:45]: Should we include a `EXPOSE` for ports 80/443 to simplify standalone executation?

File: OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    sed -i 's/LoadModule http2_module modules\/mod_http2.so/#LoadModule http2_module modules\/mod_http2.so/' /etc/httpd/conf.modules.d/10-h2.conf && \
+    sed -i 's/LoadModule proxy_http2_module modules\/mod_proxy_http2.so/#LoadModule proxy_http2_module modules\/mod_proxy_http2.so/' /etc/httpd/conf.modules.d/10-proxy_h2.conf && \
+    # Create self-signed certificate for mod_ssl
+    openssl req -x509 -nodes -newkey rsa:4096 \
+                -keyout /etc/pki/tls/private/localhost.key \
+                -out /etc/pki/tls/certs/localhost.crt \
+                -days 3650 -subj '/CN=localhost' && \
+    # Redirect logging to stdout/stderr for container logging to work
+    sed -i 's/;error_log = syslog/error_log = \/dev\/stderr/' /etc/php.ini && \
+    ln -sf /dev/stdout /var/log/httpd/access_log && \
```
Comment (mark-au) [inline OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile:41]: This will only capture HTTP not HTTPS traffic

**Review by mark-au — CHANGES_REQUESTED** [review]
Minor suggestions

File: OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile (hunk 1 above)
Comment (Djelibeybi) [reply OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile:9]: Good question. Traditionally, Apache has been held at X.Y.Z versioning and I see no indication that will change within a stream. If we get a new stream, we would have to update the module definition for libdnf anyway.

File: OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile (hunk 2 above)
Comment (Djelibeybi) [reply OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile:45]: Probably. I seem to recall I didn't add it for a specific reason but I'll need to remind myself of what that was.

File: OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile (hunk 3 above)
Comment (Djelibeybi) [reply OracleLinuxDevelopers/oraclelinux8/php/7.4-apache/Dockerfile:41]: Good catch.

**Review by mark-au — APPROVED** [review]

**Pull request merged**